src/main/java/io/confluent/connect/hdfs/HdfsSinkTask.java

```diff
@@ -58,6 +62,40 @@ public void start(Map<String, String> props) {
           throw new ConfigException("Hive Integration requires schema compatibility to be BACKWARD, FORWARD or FULL");
         }
       }
+
+      boolean secureHadoop = connectorConfig.getBoolean(HdfsSinkConnectorConfig.HDFS_AUTHENTICATION_KERBEROS_CONFIG);
+      if (secureHadoop) {
+        String principal = connectorConfig.getString(HdfsSinkConnectorConfig.HDFS_CONNECTOR_PRINCIPAL_CONFIG);
+        String keytab = connectorConfig.getString(HdfsSinkConnectorConfig.HDFS_CONNECTOR_KEYTAB_CONFIG);
+
+        final UserGroupInformation ugi = UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal,
+                                                                                      keytab);
+        log.info("Login as: " + ugi.getUserName());
+
+        final long renewPeriod = connectorConfig.getLong(HdfsSinkConnectorConfig.KERBEROS_TICKET_RENEW_PERIOD_MS_CONFIG);
+
+        isRunning = true;
+        ticketRenewThread = new Thread(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              while (isRunning) {
```

Comment:
The loop should be inside `synchronized`. If its not, you could get out of the synchronized block, continue the loop after checking `isRunning`, and then wait *after* the other thread set the flag and called `notify`. The thread would then have to wait a full renew period before exiting.

Similarly, you'll want to check the `isRunning` flag when you return from `wait` since you don't want to relogin if you're supposed to be exiting anyway (and you definitely don't want to have to wait for that renewal to complete).